functions/__fzf_search_shell_variables.fish

```diff
@@ -1,16 +1,14 @@
 function __fzf_search_shell_variables --description "Search and inspect shell variables using fzf. Insert the selected variable into the commandline at the cursor."
-    # Make sure that fzf uses fish to execute __echo_value_or_print_message, which
-    # is an autoloaded fish function so doesn't exist in other shells.
-    # Using --local so that it does not clobber SHELL outside of this function.
-    set --local --export SHELL (command --search fish)
-
     # Pipe the names of all shell variables to fzf and attempt to display the value
     # of the selected variable in fzf's preview window.
-    # Non-exported variables will not be accessible to the fzf process, in which case
-    # __echo_value_or_print_message will print an informative message in lieu of the value.
+    # Using string match to filter variables.
+    # Using string repalace to simplify output.
     set variable_name (
         set --names |
-        fzf --preview '__fzf_display_value_or_error {}'
+        fzf --preview "cat "(set --show | psub)" |
```

Comment:
Do you think it's possible to reduce calls to `set --show | psub` by calling it above and reusing the file returned by psub?

Reply:
```
❯ type psub
psub is a function with definition
# Defined in /usr/local/Cellar/fish/3.1.2/share/fish/functions/psub.fish @ line 1
function psub --description 'Read from stdin into a file and output the filename. Remove the file when the command that called psub exits.'
```

Reply:
Darn...just to confirm I also tested it :(
```
~ ❯ set file_name (echo hi | psub)
~ ❯ cat $file_name
cat: /var/folders/kv/6ynq38897hl_w48jfgzl8w080000gp/T//.psub.S7Je5fqOCy: No such file or directory
```

Reply:
> reduce calls to `set --show | psub`

You misread the code. `set --show | psub` only excutes once.